ctapipe/io/eventfilereader.py

```diff
@@ -2,36 +2,83 @@
+    Parent class for EventFileReaders of different sources.
+
+    A new EventFileReader should be created for each type of event file read
+    into ctapipe, e.g. sim_telarray files are read by the `HessioFileReader`.
+
+    EventFileReader provides a common high-level interface for accessing event
+    data from different data sources. Creating an EventFileReader for a new
+    data source ensures that data can be accessed in a common way,
+    irregardless of the data source.
+
+    EventFileReader itself is an abstract class. To use an EventFileReader you
+    must use a subclass that is relevant for the data format for the file you
+    are reader (for example you must use
+    `ctapipe.io.hessiofilereader.HessioFileReader` to read a hessio format
+    file). Alternatively you can use
+    `ctapipe.io.eventfilereader.EventFileReaderFactory` to automatically
+    select the correct EventFileReader subclass for the data format you wish
+    to read.
+
+    To loop through the events in a file:
+
+    >>> reader = EventFileReader(None, None, input_path=path)
+    >>> for event in reader:
+    >>>    print(event.count)
+
+    **NOTE**: Every time a new loop is started through the reader, it restarts
+    from the first event.
+
+    To obtain a particular event in a file:
+
+    >>> reader = EventFileReader(None, None, input_path=path)
+    >>> event = reader[event_index]
+    >>> print(event.count)
+
+    To obtain a particular event in a file from its event_id:
+
+    >>> reader = EventFileReader(None, None, input_path=path)
+    >>> event = reader["event_id"]
+    >>> print(event.count)
+
+    To obtain a slice of events in a file:
+
+    >>> reader = EventFileReader(None, None, input_path=path)
+    >>> event_list = reader[3:6]
+    >>> print([event.count for event in event_list])
+
+    To obtain a list of events in a file:
+
+    >>> reader = EventFileReader(None, None, input_path=path)
+    >>> event_indicis = [2, 6, 8]
+    >>> event_list = reader[event_indicis]
+    >>> print([event.count for event in event_list])
+
+    Alternatively one can use EventFileReader in a `with` statement to ensure
+    the correct cleanups are performed when you are finished with the reader:
+
+    >>> with EventFileReader(None, None, input_path=path) as reader:
+    >>>    for event in reader:
+    >>>       print(event.count)
 
     Attributes
     ----------
     input_path : str
-    directory : str
-        Automatically set from `input_path`.
-    filename : str
-        Name of the file without the extension.
-        Automatically set from `input_path`.
-    extension : str
-        Automatically set from `input_path`.
-    output_directory : str
-        Directory to save outputs for this file
-
+        Path to the input event file.
+    max_events : int
+        Maximum number of events to loop through in generator
     """
-    origin = None
 
     input_path = Unicode(get_dataset('gamma_test.simtel.gz'), allow_none=True,
```

Comment:
Yes sounds good

Reply:
Why are we using a class configurable trait here? I think this is the wrong approach as this would set the value for each instant of the class. I think it should just be a parameter to `__init__`

Reply:
Because these traits are how we pass command line arguments (or configuration files) to the Components.

Reply:
If one wants multiple input files in a script, one could create them as configurable traits in the script, and pass directly to the EventFileReader:
```python
reader1 = EventFileReader(self.config, self, input_path=path1)
reader2 = EventFileReader(self.config, self, input_path=path2)
```

Keeping the traitlet in EventFileReader means the user does not have to define it in the usual case where 1 file is being read with 1 reader.

Reply:
I would say this falls under a discussion of traits/components/tools and could be discussed in a different PR.